### DISCUSSION TITLE
Multidimensional partitions

### DISCUSSION THREAD

**Discussion by jtv8** [dagster-io/dagster#9703] [Q&A] [answered]
Imagine that I have a data warehouse with a set of assets that are each partitioned by time (say, daily).

Imagine also that each individual asset corresponds to a geographical region (say, continent). We've also standardised these assets so they're storing the same information in the same schema.

What I might want to do in the data warehouse is store these assets in the same physical table - e.g., a fact table for DataViz purposes.

So far so good - but let's say I want to also consider that fact table to be a _single_ asset in Dagster that's partitioned by _both_ day and continent - so that I can build downstream assets that depend on it.

Here are my questions.

 - Is there a way of achieving this with the existing tools available, or does this require a new subclass of `PartitionedConfig` called, say, `MultidimensionalPartitionedConfig`?
 - If the latter, is it possible to create a new subclass of `PartitionedConfig` in user code - or does this need to be implemented as a core feature in Dagster?
 - Does a `dynamic_partitioned_config` help here?
 - Does a `PartitionMapping` help here?

**Comment by sryza** [comment, marked as answer]
Hey @jtv8 - here's a Github issue where we're tracking this: https://github.com/dagster-io/dagster/issues/4591. How many total partitions do you imagine having?

**Reply by jtv8** [reply]
Thanks! For our use case - realistically about 10-20 static partitions multiplied by a daily partition for several years. so probably it could grow to the order of thousands to tens of thousands.

**Reply by haydenflinner** [reply]
I'd have close to a thousand partitions per day, if that changes anything :-) I see the linked ticket mentions a demo but it seems that this is now covered by https://docs.dagster.io/_apidocs/partitions#dagster.MultiPartitionsDefinition and dynamic_partitioned_config!
